libtesting-utils/src/main/java/com/mapbox/navigation/testing/ListFlow.kt

```diff
@@ -0,0 +1,87 @@
+package com.mapbox.navigation.testing
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.InternalCoroutinesApi
+import kotlinx.coroutines.flow.FlowCollector
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.withTimeout
+
+interface ListFlow<T> : StateFlow<List<T>> {
```

Comment:
Looks like all of this can be replaced with a `Channel` with unlimited buffer. It acts like a queue, where elements are added and then processed only once. 
But I even don't see it being used with coroutines, can't we have a simple `MutableList` instead?

Reply:
I need channels because the whole point was this `nextElemnts` method that would wait for the elements to come.
Thanks, I removed this custom class along with 2 gradle workarounds.